packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPluginDelegatingMethodTests.swift

```diff
@@ -0,0 +1,391 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import XCTest
+
+@testable import camera_avfoundation
+
+/// Tests of `CameraPlugin` methods delegating to `FLTCam` instance
+final class CameraPluginDelegatingMethodTests: XCTestCase {
+  private func createCameraPlugin() -> (CameraPlugin, MockFLTCam) {
+    let mockCamera = MockFLTCam()
+
+    let cameraPlugin = CameraPlugin(
+      registry: MockFlutterTextureRegistry(),
+      messenger: MockFlutterBinaryMessenger(),
+      globalAPI: MockGlobalEventApi(),
+      deviceDiscoverer: MockCameraDeviceDiscoverer(),
+      permissionManager: MockFLTCameraPermissionManager(),
+      deviceFactory: { _ in MockCaptureDevice() },
+      captureSessionFactory: { MockCaptureSession() },
+      captureDeviceInputFactory: MockCaptureDeviceInputFactory()
+    )
+    cameraPlugin.camera = mockCamera
+
+    return (cameraPlugin, mockCamera)
+  }
+
+  // Universal function orchiestrating a test of a delegating method
+  private func testDelegatingMethod(
+    // Callback with MockFLTCam instance and a callback to call when the expected FLTCam method has
+    // been called
+    setUp: (MockFLTCam, @escaping () -> Void) -> Void,
+    // Callback with CameraPlugin instance and a complition to call when the tested method finishes
+    act: (CameraPlugin, @escaping (FlutterError?) -> Void) -> Void
+  ) {
+    let (cameraPlugin, mockCamera) = createCameraPlugin()
+    // Expectation fulfilled when complition passed to `act` callback is called
+    let expectation = expectation(description: "Call completed")
+
+    var markCalled = false
+
+    setUp(
+      mockCamera,
+      {
+        markCalled = true
+      }
+    )
+
+    act(cameraPlugin) { error in
+      XCTAssertNil(error)
+      expectation.fulfill()
+    }
+
+    waitForExpectations(timeout: 30, handler: nil)
+
+    XCTAssertTrue(markCalled)
+  }
+
+  func testLockCapture_callsDelegateMethod() {
+    let targetOrientation = FCPPlatformDeviceOrientation.landscapeLeft
+
+    testDelegatingMethod { mockCamera, markCalled in
+      mockCamera.lockCaptureStub = { orientation in
+        XCTAssertEqual(orientation, targetOrientation)
+        markCalled()
+      }
+    } act: { cameraPlugin, complition in
+      cameraPlugin.lockCapture(targetOrientation, completion: complition)
+    }
+  }
+
+  func testPausePreview_callsDelegateMethod() {
+    testDelegatingMethod { mockCamera, markCalled in
+      mockCamera.pausePreviewStub = { markCalled() }
+    } act: { cameraPlugin, complition in
+      cameraPlugin.pausePreview(completion: complition)
+    }
+  }
+
+  func testPauseVideoRecording_callsDelegateMethod() {
+    testDelegatingMethod { mockCamera, markCalled in
+      mockCamera.pauseVideoRecordingStub = { markCalled() }
+    } act: { cameraPlugin, complition in
```

Comment:
if you have multiple closures, don't use trailing closure syntax. here the `setup` label is gone, making it hard to read